src/call_wrapper.ts

```diff
@@ -76,12 +112,25 @@ class CallWrapper {
         const hasArrayParameters = parameters.some((parameter) => parameter instanceof Array);
 
         if (hasArrayParameters) {
-            const memoryDescriptor = { initial: this.calculateInitialMemorySize(parameters) };
 
-            const memory = new WebAssembly.Memory(memoryDescriptor);
-            const writeableMemory = new Uint32Array(memory.buffer);
+            const tuple = CallWrapper.getMemoryLayout(parameters, currentSignature);
+
+            const memoryLayout = tuple[0];
+            fixedParameters = tuple[1];
 
-            fixedParameters = CallWrapper.fillMemory(fixedParameters, writeableMemory);
+            let maxKey = [...memoryLayout.keys()].reduce((a, b) => Math.max(a, b));
+
+            if (isOfType(memoryLayout.get(maxKey), WebAssemblyType.INT_32)) {
+                maxKey += 4;
+            } else {
```

Comment:
Use an else if here and maybe add an else part where an error gets thrown. This would be helpful when new types get introduced.

Reply:
The else branch would never be reached because `isInteger` and `isDouble` already cover all possible numbers. The map `memoryLayout` only contains numbers (or undefined which throws an error in `isOfType`)